src/main/java/seedu/address/model/activity/UniqueActivityList.java

```diff
@@ -0,0 +1,121 @@
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.model.activity.exceptions.ActivityNotFoundException;
+import seedu.address.model.activity.exceptions.DuplicateActivityException;
+
+
+/**
+ * A list of activities that enforces uniqueness between its elements and does not allow nulls.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Activity#equals(Object)
+ * @see CollectionUtil#elementsAreUnique(Collection)
+ */
+public class UniqueActivityList implements Iterable<Activity> {
+
+    private final ObservableList<Activity> internalList = FXCollections.observableArrayList();
+
+    /**
+     * Returns true if the list contains an equivalent activity as the given argument.
+     */
+    public boolean contains(Activity toCheck) {
+        requireNonNull(toCheck);
+        return internalList.contains(toCheck);
+    }
+
+    /**
+     * Adds a activity to the list.
+     *
+     * @throws DuplicateActivityException if the activity to add is a duplicate of an existing activity in the list.
+     */
+    public void add(Activity toAdd) throws DuplicateActivityException {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateActivityException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the activity {@code target} in the list with {@code editedActivity}.
+     *
+     * @throws DuplicateActivityException if the replacement is equivalent to another existing activity in the list.
+     * @throws ActivityNotFoundException if {@code target} could not be found in the list.
+     */
+    public void setActivity(Activity target, Activity editedActivity)
+            throws DuplicateActivityException, ActivityNotFoundException {
+        requireNonNull(editedActivity);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new ActivityNotFoundException();
+        }
+
+        if (!target.equals(editedActivity) && internalList.contains(editedActivity)) {
+            throw new DuplicateActivityException();
+        }
+
+        internalList.set(index, editedActivity);
+    }
+
+    /**
+     * Removes the equivalent activity from the list.
+     *
+     * @throws ActivityNotFoundException if no such activity could be found in the list.
+     */
+    public boolean remove(Activity toRemove) throws ActivityNotFoundException {
+        requireNonNull(toRemove);
+        final boolean activityFoundAndDeleted = internalList.remove(toRemove);
+        if (!activityFoundAndDeleted) {
+            throw new ActivityNotFoundException();
+        }
+        return activityFoundAndDeleted;
+    }
+
+    public void setActivity(UniqueActivityList replacement) {
```

Comment:
Travis gave an error here. I suspect that you have to put the overload methods (setActivity methods) together, rather than separating them with a remove method.